packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -179,6 +179,31 @@ export class AssociationManager {
     };
   }
 
+  private static setLaunchRoleConstraint(portfolio: IPortfolio, product: IProduct, options: CommonConstraintOptions,
+    roleProp: LaunchRoleConstraintRoleProp): void {
+    const association = this.associateProductWithPortfolio(portfolio, product, options);
+    // Check if a stackset deployment constraint has already been configured.
+    if (portfolio.node.tryFindChild(this.stackSetConstraintLogicalId(association.associationKey))) {
+      throw new Error(`Cannot set launch role when a StackSet rule is already defined for association ${this.prettyPrintAssociation(portfolio, product)}`);
+    }
+
+    const constructId = this.launchRoleConstraintLogicalId(association.associationKey);
+    if (!portfolio.node.tryFindChild(constructId)) {
+      const constraint = new CfnLaunchRoleConstraint(portfolio as unknown as cdk.Resource, constructId, {
+        acceptLanguage: options.messageLanguage,
+        description: options.description,
+        portfolioId: portfolio.portfolioId,
+        productId: product.productId,
+        ...roleProp,
```

Comment:
Very clever solution, but does specifying the properties explicitly work? I'm a little worried this trick will be difficult to follow a year from now.

```suggestion
        localRoleName: roleProp.localRoleName,
       roleArn: roleProp.roleArn,
```

Reply:
done, since one is guaranteed to be undefined, either-or-never-both should still work